src/azure-cli-core/azure/cli/core/cli_subprocess.py

```diff
@@ -0,0 +1,108 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=no-else-return
+
+import subprocess
+import re
+from knack.log import get_logger
+
+
+__all__ = [
+    # recommended invoking function
+    "run",
+    # advanced use cases
+    "CliPopen",
+    # older high-level API, can be replaced by run in many cases, support for existing codes
+    "call",
+    "check_call",
+    "check_output",
+]
+
+logger = get_logger(__name__)
+
+ARG_MASK_CHAR = re.compile(r'[$;\s]')
+
+
+def subprocess_arg_type_check(args, kwargs):
+    skip_arg_type_check = kwargs.get("skip_arg_type_check", None)
+    if skip_arg_type_check is not None:
+        del kwargs["skip_arg_type_check"]
```

Comment:
It should be 

```py
if "skip_arg_type_check" in kwargs:
    del kwargs["skip_arg_type_check"]
```

Otherwise, if `skip_arg_type_check` is `None`, it will not be deleted and passed to `subprocess`.